ogr/ogrsf_frmts/miramon/ogrmiramonlayer.cpp

```diff
@@ -0,0 +1,2613 @@
+
+void OGRMiraMonLayer::ResetReading()
+
+{
+    if (iNextFID == 0)
+        return;
+
+    iNextFID = 0;
+
+    //VSIFSeekL(m_fp, 0, SEEK_SET);
+    if (!phMiraMonLayer)
+        return;
+
+    if (phMiraMonLayer->bIsPoint && phMiraMonLayer->MMPoint.pF)
+    {
+        VSIFSeekL(phMiraMonLayer->MMPoint.pF, 0, SEEK_SET);
+        return;
+    }
+    if (phMiraMonLayer->bIsArc && !phMiraMonLayer->bIsPolygon &&
+        phMiraMonLayer->MMArc.pF)
+    {
+        VSIFSeekL(phMiraMonLayer->MMArc.pF, 0, SEEK_SET);
+        return;
+    }
+    if (phMiraMonLayer->bIsPolygon && phMiraMonLayer->MMPolygon.pF)
+    {
+        VSIFSeekL(phMiraMonLayer->MMPolygon.pF, 0, SEEK_SET);
+        return;
+    }
+}
+
+/****************************************************************************/
+/*                         GetNextRawFeature()                              */
+/****************************************************************************/
+
+void OGRMiraMonLayer::GoToFieldOfMultipleRecord(MM_INTERNAL_FID iFID,
+                                                MM_EXT_DBF_N_RECORDS nIRecord,
+                                                MM_EXT_DBF_N_FIELDS nIField)
+
+{
+    // Not an error. Simply there are no features, but there are fields
+    if (!phMiraMonLayer->pMultRecordIndex)
+        return;
+
+    fseek_function(phMiraMonLayer->pMMBDXP->pfDataBase,
+                   phMiraMonLayer->pMultRecordIndex[iFID].offset +
+                       (MM_FILE_OFFSET)nIRecord *
+                           phMiraMonLayer->pMMBDXP->BytesPerRecord +
+                       phMiraMonLayer->pMMBDXP->pField[nIField].AcumulatedBytes,
+                   SEEK_SET);
+}
+
+/****************************************************************************/
+/*                         GetNextRawFeature()                              */
+/****************************************************************************/
+
+OGRFeature *OGRMiraMonLayer::GetNextRawFeature()
+{
+    if (!phMiraMonLayer)
+        return nullptr;
+
+    if (iNextFID >= (GUInt64)phMiraMonLayer->TopHeader.nElemCount)
+        return nullptr;
+
+    OGRFeature *poFeature = GetFeature(iNextFID);
+
+    if (!poFeature)
+        return nullptr;
+
+    iNextFID++;
+    return poFeature;
+}
+
+/****************************************************************************/
+/*                         GetFeature()                                     */
+/****************************************************************************/
+
+OGRFeature *OGRMiraMonLayer::GetFeature(GIntBig nFeatureId)
+
+{
+    OGRGeometry *poGeom = nullptr;
+    OGRPoint *poPoint = nullptr;
+    OGRLineString *poLS = nullptr;
+    MM_INTERNAL_FID nIElem = (MM_INTERNAL_FID)nFeatureId;
+    MM_EXT_DBF_N_MULTIPLE_RECORDS nIRecord = 0;
+
+    if (!phMiraMonLayer)
+        return nullptr;
+
+    if (nFeatureId < 0)
+        return nullptr;
+
+    // In case of nIElem == 0 and there are no elements we give the chance
```

Comment:
I don't get this. For an empty layer, there is no feature to return. I'm probably missing what you mean

Reply:
When I was importing an empty SHP (no features but with an empty DBF with some empty fields) I realized that it would be nice to obtain an empty layer and get the structure of the DBF (with the name of the fields... etc.).

MOre clear now?

Reply:
I understand the aim, but you don't need to return a dummy feature for that. A layer with 0 features is perfectly fine and can be explored with the OGR API and tools

Reply:
Ok, then I leave it as it should be to avoid corrupted acces.

if (nIElem >= phMiraMonLayer->TopHeader.nElemCount)
        return nullptr;